### PR TITLE
2934 make compressed http2 request possible

### PR DESCRIPTION
**Pull request by KowalczykBartek**
Signed-off-by: Kowalczyk <bkowalczyyk@gmail.com>

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/Http1xUpgradeToH2CHandler.java [hunk 1]
```diff
@@ -26,10 +27,12 @@
   private final HttpServerChannelInitializer initializer;
   private final HandlerHolder<? extends Handler<HttpServerConnection>> holder;
   private VertxHttp2ConnectionHandler<Http2ServerConnection> handler;
+  private HttpServerOptions httpServerOptions;
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/Http1xUpgradeToH2CHandler.java]: rather use 2 booleans for compression/decompression supported than the full options

**Comment by vietj** [comment]
looks good, minor change asked

File: src/main/java/io/vertx/core/http/impl/Http1xUpgradeToH2CHandler.java (hunk 1 above)
Comment (KowalczykBartek) [reply src/main/java/io/vertx/core/http/impl/Http1xUpgradeToH2CHandler.java]: fixed

**Comment by vietj** [comment]
thanks for the contribution

**Pull request merged**